pkg/kubelet/status/status_manager.go

```diff
@@ -854,6 +864,16 @@ func (m *manager) canBeDeleted(pod *v1.Pod, status v1.PodStatus) bool {
 	if pod.DeletionTimestamp == nil || kubetypes.IsMirrorPod(pod) {
 		return false
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// Delay deletion of Pending pods until the phase is terminal, which
+		// is expected to be set in the TerminatePod function. The check for
+		// presence of a finalizer is optimization to avoid additional pod
```

Comment:
What's the motivation for this optimization?  Do we have a measurement that it is necessary?  In general we want the pod lifecycle to be consistent and not dependent on conditionals.  Kubelets write lots of excessive status events inefficiently anyway and we plan to deal with that holistically - I tend to doubt this optimization is necessary or desirable in the long run.

Reply:
No hard data, it was just to make sure we wouldn't worsen performance if the end state does not matter, given small QPS for Kubelet. But,
- the QPS will be increased in the release: https://github.com/kubernetes/kubernetes/pull/116121
- it is cleaner
- the solution without the finalizer check was approved in KEP.

Thus, I agree with the sentiment, I will make the simplification.

Reply:
Done